### PR TITLE
ICMSLST-2233 Exclude specific users from data migration

### PR DESCRIPTION
**Pull request by mikeylockwood**
- Exclude users with username not an email address from the data migration
- Exclude users with specific email domain from the data migration
- Add `icms_v1_user` flag to migrated users

### CODE REVIEW

File: config/env.py [hunk 1]
```diff
@@ -123,6 +123,7 @@ class Environment(BaseSettings):
     icms_v1_replica_dsn: str = ""
     icms_prod_user: str = ""
     icms_prod_password: str = ""
+    data_migration_email_domain_exclude: str = ""
```
Comment (MattHolmes123) [inline config/env.py:126]: Do you only ever plan to support ignoring a single domain or could the list grow in future?

File: data_migration/models/user/access.py [hunk 2]
```diff
@@ -56,6 +59,19 @@ def data_export(cls, data: dict[str, Any]) -> dict[str, Any]:
...
         return data
 
+    @classmethod
+    def get_source_data(cls) -> Generator:
+        values = cls.get_values()
+        values_kwargs = cls.get_values_kwargs()
+        related = cls.get_related()
+        return (
+            cls.objects.select_related(*related)
+            .exclude(submitted_by__username__endswith=EXCLUDE_DOMAIN)
```
Comment (MattHolmes123) [inline data_migration/models/user/access.py:69]: `iendswith` for case insensitive search (same for all places you've used it)

File: data_migration/queries/permissions.py [hunk 3]
```diff
@@ -57,6 +62,8 @@
   END login_id
   FROM securemgr.web_user_account_histories uah
   WHERE uah.person_id_current IS NOT NULL
+    AND login_id LIKE '%@%'
+    AND login_id NOT LIKE '%{EXCLUDE_DOMAIN}'
```
Comment (MattHolmes123) [inline data_migration/queries/permissions.py:66]: `ILIKE`

File: scripts/reset-data-migrations [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+set -e
+
+rm data_migration/migrations/*.py
+make data_migrations
+
+# Restore the data migrations
+git restore data_migration/migrations/0002_data_firearms_act.py
+
+make black_format
+make isort_format
```
Comment (MattHolmes123) [inline scripts/reset-data-migrations:12]: Missing newline

**Review by MattHolmes123 — APPROVED** [review]

**Comment by MattHolmes123** [comment]
Have you done the other aspects of ICSMLST-2234 already?
image

File: config/env.py (hunk 1 above)
Comment (mikeylockwood) [reply config/env.py:126]: Just a single domain it looks like. If needed in future I can change it to a list

**Pull request merged**